pkg/kv/kvtest/store.go

```diff
@@ -0,0 +1,253 @@
+package kvtest
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/treeverse/lakefs/pkg/kv"
+)
+
+type makeStore func(t *testing.T, ctx context.Context) kv.Store
+
+func TestDriver(t *testing.T, name string, dsn string) {
+	ctx := context.Background()
+	ms := makeStoreByName(name, dsn)
+	t.Run("Driver_Open", func(t *testing.T) { testDriverOpen(t, ctx, ms) })
+	t.Run("Store_SetGet", func(t *testing.T) { testStoreSetGet(t, ctx, ms) })
+	t.Run("Store_SetIf", func(t *testing.T) { testStoreSetIf(t, ctx, ms) })
+	t.Run("Store_Delete", func(t *testing.T) { testStoreDelete(t, ctx, ms) })
+	t.Run("Store_Scan", func(t *testing.T) { testStoreScan(t, ctx, ms) })
+}
+
+func testDriverOpen(t *testing.T, ctx context.Context, ms makeStore) {
+	store1 := ms(t, ctx)
+	store1.Close()
+
+	store2 := ms(t, ctx)
+	store2.Close()
+}
+
+func testStoreSetGet(t *testing.T, ctx context.Context, ms makeStore) {
+	store := ms(t, ctx)
+	defer store.Close()
+
+	testKey := []byte("key")
+	testValue1 := []byte("value1")
+	testValue2 := []byte("value2")
+
+	// set test key with value1
+	err := store.Set(ctx, testKey, testValue1)
+	if err != nil {
+		t.Fatalf("failed to set key '%s', to value '%s': %s", testKey, testValue1, err)
+	}
```

Comment:
I like `require.NoError(...)` better as a one liner 0 indentation check..

Reply:
May take this into consideration - all Go implemented without 'require' so I'm still fine with that - unless more people can't stand this code, and I'll replace.